### PR TITLE
Fix issue #48161 - brew doctor check for orphaned python directories.

### PR DESCRIPTION
**Pull request by tzickel**

### CODE REVIEW

File: Library/Homebrew/diagnostic.rb [hunk 1]
```diff
@@ -1347,6 +1347,31 @@ def check_for_external_cmd_name_conflict
...
+          if major == '2'
+            if Dir["#{HOMEBREW_CELLAR}/python/#{ver}.*"].empty?
+              mayberemove.push(x)
+            end
+          elsif major == '3'
+            if Dir["#{HOMEBREW_CELLAR}/python3/#{ver}.*"].empty?
+              mayberemove.push(x)
+            end
+          end
+        end
```
Comment (tdsmith) [inline Library/Homebrew/diagnostic.rb]: This is similar and a little more succinct:

```ruby
Dir["#{HOMEBREW_PREFIX}/lib/python*/site-packages"].select do |path|
	match = /python((\d)\.\d+)/.match path
	next unless match
	xy, major = match[1..2]
	formula_name = "python#{major == "2" ? "" : major}"
	Dir[HOMEBREW_CELLAR/formula_name/"#{xy}.*"].empty?
end
```

Comment (tdsmith) [inline Library/Homebrew/diagnostic.rb]: updated to add `mayberemove = `

**Comment by DomT4** [comment]
@tdsmith Still interested in this?

**Comment by apjanke** [comment]
Closing this due to inactivity/lack of response. If you'd still like to pursue this, feel free to respond to the previous comments/questions and re-open on the new [`homebrew/brew` repo](https://github.com/homebrew/brew).

**Pull request closed without merging**